test/nodetool/conftest.py

```diff
@@ -27,31 +27,51 @@ def pytest_addoption(parser):
                      " with --nodetool=cassandra, this should be the nodetool binary")
     parser.addoption('--jmx-path', action='store', default=None,
                      help="Path to the jmx binary, only used with --nodetool=cassandra")
+    parser.addoption('--run-within-unshare', action='store_true',
+                     help="Setup the 'lo' network if launched with unshare(1)")
 
 
 @pytest.fixture(scope="session")
-def rest_api_mock_server():
-    ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
-    port = random.randint(10000, 65535)
+def maybe_setup_loopback_network(request):
+    # unshare(1) -rn drops us in a new network namespace in which the "lo" is
+    # not up yet, so let's set it up first.
+    if request.config.getoption('--run-within-unshare'):
+        try:
+            args = "ip link set lo up".split()
+            subprocess.run(args, check=True)
+        except FileNotFoundError:
+            args = "/sbin/ifconfig lo up".split()
+            subprocess.run(args, check=True)
+    yield
 
-    server_process = subprocess.Popen([
-        sys.executable,
-        os.path.join(os.path.dirname(__file__), "rest_api_mock.py"),
-        ip,
-        str(port)])
 
+@pytest.fixture(scope="session")
+def rest_api_mock_server(request, maybe_setup_loopback_network):
+    ip = '127.0.0.1'
```

Comment:
Would be nice if this code works, as I beleive it did before (see README.md), also outside of test.py.
So if you are NOT using the new "unshare" option, shouldn't you leave the old random IP and/or port intact?

Reply:
good point. actually, i always run the tests using pytest or directly as long as i can. i will add the random IP/port back if the unshare option is not enabled. will do so early tomorrow.